app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -920,15 +920,40 @@ def configure_vnic_with_requested_data(name, requested_vnic, destination_vnic, n
     end
 
     def private_load_allowed_networks(vlans, uid_ems_cluster)
-      profiles = get_vnic_profiles_in_cluster(uid_ems_cluster)
+      profiles = local_get_vnic_profiles_in_cluster(uid_ems_cluster)
       profiles.each do |profile, profile_network|
-        vlans[profile.id] = "#{profile.name} (#{profile_network.name})"
+        vlans[profile.uid_ems] = "#{profile.name} (#{profile_network.name})"
       end
 
       vlans['<Empty>'] = _('<No Profile>')
       vlans['<Template>'] = _('<Use template nics>')
     end
 
+    def local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      cluster_profiles = {}
+      profiles = all_vnic_profiles_as_lans_in_ems(ext_management_system)
+      cluster_networks = all_networks_as_switches_in_cluster(uid_ems_cluster)
+      profiles.each do |p|
+        profile_network = cluster_networks.detect { |n| n.id == p.switch.id }
+        if profile_network
+          cluster_profiles[p] = profile_network
+        end
+      end
+      cluster_profiles
+    end
+
+    def all_networks_as_switches_in_cluster(uid_ems_cluster)
+      cluster = EmsCluster.find_by(:uid_ems => uid_ems_cluster)
+      hosts = cluster.hosts
+      host_switches = hosts.collect(&:host_switches).flatten.compact.uniq
+      host_switches.collect(&:switch).flatten.compact.uniq
+    end
+
+    def all_vnic_profiles_as_lans_in_ems(ems)
+      ManageIQ::Providers::InfraManager::DistributedVirtualSwitch.where(ems_id: ems.id)
```

Comment:
Can we define an Ovirt subclass of `DistributedVirtualSwitch` like `ManageIQ::Providers::Ovirt::InfraManager::DistributedVirtualSwitch`?  Also could you use `ems.distributed_virtual_switches` instead of `DistributedVirtualSwitch.where(:ems_id => ems.id)` ?

Reply:
done